File: packages/dvai-bridge-android/android/src/main/java/co/deepvoiceai/bridge/BoundServer.kt
package co.deepvoiceai.bridge

import co.deepvoiceai.bridge.license.LicenseStatus

/**
 * Result of a successful [DVAIBridge.start] call. Mirrors the iOS DVAIBridge
 * `BoundServer` struct + the Capacitor JS shim's StartResult.
 *
 * @param baseUrl       Full base URL of the embedded OpenAI-compatible server,
 *                      including the `/v1` suffix. Example:
 *                      `"http://127.0.0.1:38883/v1"`.
 * @param port          Port the HTTP server actually bound to (port-fallback may
 *                      have moved it past [StartOptions.httpBasePort]).
 * @param backend       The backend that actually loaded — useful when [StartOptions]
 *                      set [BackendKind.Auto] and the consumer wants to know what
 *                      was picked.
 * @param modelId       Stable identifier for the loaded model. Surfaced in the
 *                      `model` field of every OpenAI response.
 * @param licenseStatus v3.3 — outcome of the offline JWT license check that
 *                      ran during [DVAIBridge.start]. Null when the validator
 *                      didn't run (legacy callers, missing context). Production
 *                      starts always carry a non-null status here; failure modes
 *                      throw before this struct is produced.
 */
data class BoundServer(
    val baseUrl: String,
    val port: Int,
    val backend: BackendKind,
    val modelId: String,
    val licenseStatus: LicenseStatus? = null,
)

/** Read-only status snapshot returned by [DVAIBridge.status]. */
data class StatusInfo(
    val running: Boolean,
    val baseUrl: String? = null,
    val backend: BackendKind? = null,
    val modelId: String? = null,
)

File: packages/dvai-bridge-dotnet/src/DVAIBridge/BoundServer.cs
namespace DVAIBridge;

/// <summary>
/// The bound, running HTTP server returned by
/// <see cref="DVAIBridge.StartAsync(StartOptions, System.Threading.CancellationToken)"/>.
/// </summary>
/// <param name="BaseUrl">
/// Full base URL ending in <c>/v1</c> — e.g. <c>http://127.0.0.1:38883/v1</c>.
/// Pass this to any OpenAI-compatible .NET HTTP client.
/// </param>
/// <param name="Port">The TCP port the server is listening on.</param>
/// <param name="Backend">Resolved backend (never <see cref="BackendKind.Auto"/>).</param>
/// <param name="ModelId">
/// Identifier surfaced via <c>/v1/models</c>. Defaults to the model filename
/// when <see cref="StartOptions.ModelId"/> was not supplied.
/// </param>
/// <param name="LicenseStatus">
/// The license validation outcome produced at startup, or <c>null</c>
/// when license validation was not run (legacy code paths / test
/// fakes that bypass the facade's validator step).
/// <c>Commercial</c> / <c>Trial</c> = paid; <c>FreeDev</c> = developer
/// bypass; <c>FreeProd</c> / <c>FreeExpired</c> never reach here in
/// production because the facade's strict
/// <see cref="License.LicenseValidator.ValidateAndAssertAsync(System.Threading.CancellationToken)"/>
/// throws before <c>StartAsync</c> returns.
/// </param>
public sealed record BoundServer(
    string BaseUrl,
    int Port,
    BackendKind Backend,
    string ModelId,
    License.LicenseStatus? LicenseStatus = null);
